codalab/rest/interpret.py

```diff
@@ -608,5 +610,25 @@ def expand_raw_item(raw_item):
                 raw_items.append(subworksheet_item(worksheet_info) + (None, None))
 
         return raw_items
+    else:
+        raise Exception("not a search query")
+
+
+def expand_search_items(raw_items):
+    return list(chain.from_iterable([expand_search_item(raw_item) for raw_item in raw_items]))
+
+
+def expand_search_item(raw_item):
+    """
+    Raw items that include searches must be expanded into more raw items.
+    Input: Raw item.
+    Output: Array of raw items. If raw item does not need expanding,
+    this returns an 1-length array that contains original raw item,
+    otherwise it contains the search result. You do not need to call
+    resolve_items_into_infos on the returned raw_items.
+    """
+    (bundle_info, subworksheet_info, value_obj, item_type, id, sort_key) = raw_item
+    if item_type == TYPE_DIRECTIVE and get_command(value_obj) in ('search', 'wsearch'):
```

Comment:
Here we check if a raw item is either 'search' or 'wsearch', but in line 177, we don't check the conditions. I think we should be consistent in a way either: calling perform_search_query requires checking get_command(value_obj) in ('search', 'wsearch')  or it's safe to call it even if the input is not a search object. If this makes sense

Reply:
I see what you mean, but in this case, we have to check if `get_command(value_obj) in ('search', 'wsearch')` because there are some directives that don't correspond to search queries. If we were to move the `get_command(value_obj) in ('search', 'wsearch')` logic into `perform_search_query`, then we would be calling `perform_search_query` even on non-search directives.

Currently, it's safe to call `perform_search_query` even if the input is not a search object -- it's just that we don't _want_ to call it (we want to instead just return `[raw_item]`) when the input is not a search object.